### PR TITLE
#3354 #3377

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#3354 Graphic window appears every time I start Visual Studio
- suppress `utils::win*` on remote and non-interactive

#3377 Add database connection doesn't handle AAD
- improve SQL/ODBC conversion
- detect AAD on Azure
- Use ODBC 13 driver for AAD/Azure
- prompt to install new ODBC driver if missing
- tests

### CODE REVIEW

File: src/Host/Client/Impl/Session/RSession.cs [hunk 1]
```diff
@@ -488,6 +496,19 @@ public Task StartHostAsync(RHostStartupInfo startupInfo, IRSessionCallback callb
 "));
         }
 
+        private static Task SuppressUI(IRExpressionEvaluator eval) {
+            // # Suppress Windows UI 
+            // http://astrostatistics.psu.edu/datasets/R/html/utils/html/winMenus.html
+            return eval.ExecuteAsync(@"
+                rtvs:::suppress_function('bringToTop', 'grDevices');
```
Comment (karthiknadig) [inline src/Host/Client/Impl/Session/RSession.cs]: Can we do this entirely in R? as in rtvs:::suppress_all(), this way we don'y need to eval the replacement function each time.

Comment (int19h) [inline src/Host/Client/Impl/Session/RSession.cs]: There's only one eval here, it just calls the function multiple times.

Comment (karthiknadig) [inline src/Host/Client/Impl/Session/RSession.cs]: i meant this eval in the R code:
 replacement <- eval(parse(text = "function(...) { stop('Not supported') }"));

File: src/Host/Client/Impl/rtvs/R/overrides.R [hunk 2]
```diff
@@ -82,3 +82,25 @@ remove.packages <- function(...) {
     utils::remove.packages(...)
     invisible(rtvs:::send_notification('!PackagesRemoved'))
 }
+
+suppress_function <- function(function_name, package_name) {
+    package_spec <- paste("package:", package_name, sep='');
+    # The message may only become visible in remote session.
+    # TODO: provide localized replacement.
+    replacement <- eval(parse(text = "function(...) { stop('Not supported') }"));
```
Comment (int19h) [inline src/Host/Client/Impl/rtvs/R/overrides.R]: I don't think you need eval here, and you definitely don't need parse. If eval is needed, you can use quote:
```r
replacement <- eval(quote(function(...) { stop('Not supported') }))
```
But I don't see the reason for eval, either. You should just be able to use the function literal directly.

**Comment by int19h** [comment]
👍

**Pull request merged**